### PR TITLE
feat(layout-direction): a service to accompany the Dir directive

### LINKED ISSUE #1719: Introduce [ltr, rtl] layout direction service
Similar to what the Dir directive does, but will also capture "dir" attributes outside the Angular app. 

cc: @jelbourn

### PR DESCRIPTION
**Pull request by EladBezalel**
- Finds the application root element and traverses upwards and looking for dir attribute
- If Dir directive is available throughout the application it will be injected instead the activeLayoutDirection service

fixes #1719

### CODE REVIEW

File: src/lib/core/rtl/active-layout-direction.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  ElementRef,
+  ApplicationRef,
+  Injector
+} from '@angular/core';
+
+import {LayoutDirection} from './dir';
+import {EventEmitter} from '@angular/common/src/facade/async';
+
+@Injectable()
+export class ActiveLayoutDirection {
```
Comment (jelbourn) [inline src/lib/core/rtl/active-layout-direction.ts]: Rename this class to something like `GlobalDirAccessor` and then add another class called `ActiveLayoutDirection` that just looks like
```ts
export class ActiveLayoutDirection {
  dirChange: Observable<void>;
  getActiveDirection(injector: Injector): Promise<LayoutDirection> { return null; }
}
```

The both `Dir` and `GlobalDirAccessor` will have `implements ActiveLayoutDirection` (TypeScript lets you `implement` a class just like an interface. Keeping it as a class lets us use it as an injection token).

File: src/lib/core/rtl/active-layout-direction.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import {
+  Injectable,
+  ElementRef,
+  ApplicationRef,
+  Injector
+} from '@angular/core';
+
+import {LayoutDirection} from './dir';
+import {EventEmitter} from '@angular/common/src/facade/async';
```
Comment (EladBezalel) [inline src/lib/core/rtl/active-layout-direction.ts]: Webstorm auto import 😅

**Comment by jelbourn** [comment]
@EladBezalel was this ready for another round of review?

**Comment by EladBezalel** [comment]
nope 😅

**Comment by EladBezalel** [comment]
A better solution was suggested on #3600

**Pull request closed without merging**